### PR TITLE
Continuous deployment

### PR DESCRIPTION
**Pull request by davidroberts63**
Hookup to the continuous deployment process I spoke about in my talk.

Note that at the moment you won't get the super duper fancy screenshots in this pull request until CircleCI can follow this repository. In order to get that going add me (@davidroberts63) as a committer to this repository. Then I'll tell CircleCI to follow techlahoma/okcjs-website. Boom screenshots ready to go as well as CD.

If you're nervous about granting commit rights that's okay. I just need rights to add web hooks to okcjs-website repository for CircleCI to see it. If my account doesn't have commit access then screenshots should work but continuous deployment won't.

You can test this out locally. Just make sure to have PhantomJS installed. Run `hexo server` then `phantomjs screenshot.js` in separate shells. Open the images in the screenshots folder.

Once that's working I'm going to update Suede-Halibut to be able to use individual security tokens (or something else) to commit and update PRs. So we don't have my access showing up everywhere.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -24,10 +24,6 @@ permalink_defaults:
 # Directory
 source_dir: source
 public_dir: public
-tag_dir: tags
-archive_dir: archives
-category_dir: categories
-code_dir: downloads/code
```
Comment (davidroberts63) [inline _config.yml]: Removed these because they are duplicated above.

**Comment by the-simian** [comment]
Oh this is cool.

**Comment by the-simian** [comment]
I'm still interested in making this a module, but I've been deep in the TP2015 videos. I am close to done - I have to render one and edit and render another and that's all she wrote

**Comment by DevinClark** [comment]
Would it be better for us to use Codeship for this? If I remember correctly, they gave us a free plan on that?

**Comment by davidroberts63** [comment]
That can be setup with minimal changes I'm not partial to CircleCI, it's just what I used for OKCsharp. CircleCI does provide for a free single container build process.  https://circleci.com/pricing/

So unless we feel that there will be a lot of pushes I imagine we will be fine.

**Comment by the-simian** [comment]
I've used travis before, but im *also* not stuck on any particular service. I think travis was just very popular for a while, and I used it in my open source stuffs

**Comment by the-simian** [comment]
is Circle free for OSS stuff?

**Comment by davidroberts63** [comment]
It's pricing is based on build containers. The first one is free for any number of repositories and users up to 1,500 builds per month. Doesn't mention OSS specifically.

Quick look on Codeship they don't store build artifacts (outside of the build log) so we'd have to modify the process a bit to upload the screenshots to S3 or something similar.

**Comment by the-simian** [comment]
ok, let me check out travis so we can compare. my coworker @joeharedotorg has one he worked with recently he really likes also, I'll ask him about that

**Comment by the-simian** [comment]
Codeship:

codehsip-insides

**Comment by the-simian** [comment]
and here is travis: they recently  improved the ui a lot
travis-download

**Comment by the-simian** [comment]
there is also https://jenkins-ci.org/

**Comment by jhare** [comment]
@davidroberts63 I use wercker.com and am pretty happy with it. Docker-based containers, build and deploy functionality. The site is slick and they have a sort of "community" section where you can see what other people are doing in their `yml` to build and deploy.

**Comment by the-simian** [comment]
For the lazy: [wercker.com/](http://wercker.com/)

**Comment by the-simian** [comment]
FYI: this is an npm module that just came out I have been experimenting with also : https://github.com/sindresorhus/pageres

**Comment by davidroberts63** [comment]
Here's the CircleCI one. You can easily see it yourself I think, just login to CircleCI (it'll auth via Github) and you should be able to see the same if I am thinking corerctly.:
image

Woh. I did a copy and paste of the image and Github just put it there. That was really cool.

**Comment by davidroberts63** [comment]
Would we like to try to set this up in all of them and do a comparison? That would be fun.

**Comment by davidroberts63** [comment]
Sort of a Hosted CI smash down

**Comment by davidroberts63** [comment]
I saw the pageres npm package earlier @the-simian. I had planned on trying to use that but just went with what I had as a learning process. I would very much like to use pageres though. It's very easy to read.

Still need to use PhantomJS itself in order to get the individual post url or any others that we may want to render.

**Comment by davidroberts63** [comment]
@jhare I like the idea of that wrecker community. I'll take a look.

**Comment by the-simian** [comment]
> Would we like to try to set this up in all of them and do a comparison? That would be fun.

Hell. Yes. Why not? They're all free. I set up more than one on another module and it wasn't a big whoop. I guess the biggest concern is - I guess hwo each one handles accounts and permissions. can I go log in as me, make an okcjs project and give you, jess, amanda, vance etc permission to go fiddle with it? one wayt o find out

in the SUNDAY NIGHT RAW CONTINUOUS INTEGRATION ULTIMATE 

**SMAAAACKDOWN**

**Comment by davidroberts63** [comment]
Have not forgotten about this. I'm going to spend some time this weekend perhaps trying out the other CI tools on my hexo-test repository.

**Comment by the-simian** [comment]
@davidroberts63 cool I've been somewhat distracted by some work in redux + RxJs (which might become a lightning talk)

I'n nearly done though, but lmk how your experiments go

**Pull request merged**